keys/types.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import "github.com/coinbase/rosetta-sdk-go/types"
+
+// PrivateKey contains the hex-encoded privkey bytes as well as the CurveType
+type PrivateKey struct {
+	Bytes     []byte
```

Comment:
@juliankoh thoughts on making this `hex` so that we could store as JSON. If we use bytes here, we can't store as easily.

Reply:
@patrick-ogrady bytes serializes to JSON just fine in Go, except it in base64, so the hex standardization choice makes things more difficult 😛

Reply:
I should've dug more deeply into this topic before writing this comment @jakecraige, my bad.

@juliankoh let's change this to use `[]byte` again but use the custom marshaler I wrote (in #60) to serialize as hex instead of base64 to be consistent with other byte representations in the SDK.

Here is an example of the custom marshaling logic I use: https://github.com/coinbase/rosetta-sdk-go/blob/cd4a060d72c34ce5391198e65f3bdb42890fa7dd/types/public_key.go#L31-L70